contracts/ExampleOracleSimple.sol

```diff
@@ -2,11 +2,11 @@ pragma solidity =0.6.6;
 
 import '@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol';
 import '@uniswap/v2-core/contracts/interfaces/IUniswapV2Pair.sol';
+import '@uniswap/lib/contracts/libraries/FixedPoint.sol';
```

Comment:
what is this?

Reply:
https://github.com/Uniswap/uniswap-lib/blob/master/contracts/libraries/FixedPoint.sol

Reply:
Thanks ✨🙏. 

Is `@uniswap/lib` currently used in any other uniswap contracts? I’m asking because I think in this context the cost of indirection (on auditability etc.) is often much higher than the cost of duplication, and I’m not sure that extracting these libraries to a separate repo is the right trade off to make.

Reply:
When we copy-paste code we risk:

- code integrity, how do we know it hasn't been modified?
- unit test coverage, we do not also copy paste the unit tests of these libraries
- inconsistencies == poor usability (see core vs. periphery for these libraries)

Using a package manager like yarn/npm, we get integrity via lockfiles and integrity hashes. We unit test all library code separately, and it can be audited for correctness and security once per release. Inconsistencies will not be an issue assuming we follow semantic versioning.

Re: auditability, the build folder contains all built contract bytecode and source maps, including from other packages. Automated tooling is not affected.

Manual auditability via GitHub would benefit from better linking of imports between repos and within the same repo. When coding however, most IDEs that support solidity will allow ctrl/cmd-click to see the source with zero overhead.